src/globals.js

```diff
@@ -1,18 +1,34 @@
 import { debug } from './logger';
 
 export const SpansContainer = (() => {
-  const spansToSend = [];
+  let spansToSend = {};
 
   const addSpan = span => {
-    spansToSend.push(span);
+    spansToSend[span.id] = span;
     debug('Span created', span);
   };
-  const getSpans = () => spansToSend;
-  const clearSpans = () => (spansToSend.length = 0);
+  const getSpans = () => Object.values(spansToSend);
+  const clearSpans = () => (spansToSend = {});
 
   return { addSpan, getSpans, clearSpans };
 })();
 
+export const GlobalTimer = (() => {
+  let currentTimer = undefined;
+
+  const setGlobalTimeout = (func, duration) => {
+    clearTimer();
+    currentTimer = setTimeout(func, duration);
+    currentTimer.unref();
```

Comment:
why is this needed? isn't `clearTimer` going to take care of it? is this here "just in case"?